rpc/grpc/listen_and_serve.go

```diff
@@ -34,8 +37,20 @@ func FromExistingServer(listenAndServe ListenAndServe) *Plugin {
 }
 
 // ListenAndServeGRPC starts a netListener.
-func ListenAndServeGRPC(config *Config, grpcServer *grpc.Server) (netListener net.Listener, err error) {
-	netListener, err = net.Listen("tcp", config.Endpoint)
+func ListenAndServeGRPC(config *Config, grpcServer *grpc.Server, log logging.Logger) (netListener net.Listener, err error) {
+	// Start listener with TCP/IP or unix domain socket type
+	if config.Endpoint != "" {
+		netListener, err = net.Listen("tcp", config.Endpoint)
+		log.Info("Listening GRPC on tcp://", config.Endpoint)
+	} else if config.UnixSocketFilePath != "" {
+		// Remove old socket
+		syscall.Unlink(config.UnixSocketFilePath)
+		// Socket file is created automatically
+		netListener, err = net.Listen("unix", config.UnixSocketFilePath)
```

Comment:
What if we don't own the socket file (which will not always be the case).  Removing it is kind of aggressive.